xpdan/data_reduction.py

```diff
@@ -272,20 +289,28 @@ def integrate_and_save(headers, dark_sub_bool=True,
                 f_name = 'sum_' + rest[-1] + f_name
 
             # masking logic
+            # workflow for xpdAcq v0.5.1 release, will change later
             mask = np.ones(img.shape).astype(bool)
-            if auto_mask:
-                print("INFO: mask your image: {}".format(f_name))
-                f_name = 'masked_' + f_name
-                if mask_dict is None:
-                    mask_dict = an_glbl.mask_dict
-                dummy_img = np.copy(img)  # prepare for masking
-                dummy_img /= ai.polarization(dummy_img.shape,
-                                             polarization_factor)
-                mask = mask_img(dummy_img, ai, **mask_dict)
+            if mask=='default':
+                mask_md = header.start.get('mask', None)
+                if mask_md is None:
+                    print("INFO: no mask associated or mask information was"
+                          " not set up correctly, no mask will be applied")
+                # unpack here 
+                data, ind, indptr = mask_md
+                print("INFO: pull off mask associate with your image: {}"
+                      .format(f_name))
+                mask = decompress_mask(data, ind, indtpr, img.shape)
+            elif mask=='auto':
+                mask = mask_img(img, ai, **glbl.mask_dict)
+            elif mask=='None':
+                mask = None
+            mask_fn = os.path.splitext(f_name)[0]  # remove ext
+            if mask is not None:
                 print("INFO: mask file '{}' is saved at {}"
-                      .format(f_name, root_dir))
-                np.save(os.path.join(root_dir, f_name),
-                        mask)
+                      .format(mask_fn, root_dir))
+                np.save(os.path.join(root_dir, mask_fn),
+                        ~mask)  # default is .npy from np.save
```

Comment:
As user want to visualize their mask with existing plotting program, which follows different convention. Current mask is different from the convention with ``pyFAI`` and ``fit2D``. 
should we rethink about the convention?

@sbillinge  any comment?